mm/2s2h/Enhancements/Cutscenes/StoryCutscenes/SkipLearningGoronLullabyIntro.cpp

```diff
@@ -0,0 +1,56 @@
+#include <libultraship/bridge.h>
+#include "2s2h/GameInteractor/GameInteractor.h"
+#include "2s2h/CustomMessage/CustomMessage.h"
+#include "2s2h/CustomItem/CustomItem.h"
+#include "2s2h/Rando/Rando.h"
+#include "2s2h/ShipInit.hpp"
+
+extern "C" {
+#include "variables.h"
+#include "functions.h"
+#include "overlays/actors/ovl_En_Jg/z_en_jg.h"
+}
+
+#define CVAR_NAME "gEnhancements.Cutscenes.SkipStoryCutscenes"
+#define CVAR CVarGetInteger(CVAR_NAME, 0)
+
+void RegisterSkipLearningGoronLullabyIntro() {
+    COND_VB_SHOULD(VB_JG_THINK_YOU_KNOW_LULLABY, CVAR || IS_RANDO, {
+        EnJg* ObjectActor = va_arg(args, EnJg*);
+        Player* player = GET_PLAYER(gPlayState);
+
+        // Always consider lullaby known so we don't go into the cutscene to learn it
+        *should = true;
+
+        // Goron Elder sets this reg at the end of the cutscene
+        // Manually set it here as it is skipped
+        SET_WEEKEVENTREG(WEEKEVENTREG_24_40);
+
+        if (IS_RANDO) {
+            if (!RANDO_SAVE_CHECKS[RC_PATH_TO_GORON_VILLAGE_LULLABY_INTRO].obtained) {
+                RANDO_SAVE_CHECKS[RC_PATH_TO_GORON_VILLAGE_LULLABY_INTRO].eligible = true;
+            }
+        } else {
```

Comment:
While it is usually annoying, we want these concepts (skips/rando) isolated from each other. The `IS_RANDO` usage outside of the `Rando/` folder should be minimal, and we shouldn't be setting eligible status here.

Reply:
Understood. I found the pattern I'm looking for that will solve this issue in SkipLearningSonataOfAwakening, wherein:
- The cutscene skip defines a VB Should hook on the item grant during the cutscene that the Rando function can override.
- The rando actor behavior can use that hook to prevent the vanilla song/item from being given, and instead grant the rando check reward.